proposed/0006-Launcher.md

```diff
@@ -0,0 +1,204 @@
+```
+[
+    {
+        App Name,
+        Reference Count,
+        Random Unique App ID,    // 64 Bytes
+        Vec<SHA512(App-Binary)>, // Also serves to tell how many machines where the App is added to Launcher
+        <APP-ROOT-DIR-Key>,
+        Option<SAFEDrive-Directory-Key>,
+        OtherMetadata, // For Future Use
+    }
+    {
+        App Name,
+        Reference Count,
+        Random Unique App ID,    // 64 Bytes
+        Vec<SHA512(App-Binary)>, // Also serves to tell how many machines where the App is added to Launcher
+        <APP-ROOT-DIR-Key>,
+        Option<SAFEDrive-Directory-Key>,
+        OtherMetadata, // For Future Use
+    },
+    … etc
+]
+```
+- Create/Append to local config file (on the user's machine) the following:
+```
+[
+    { App-ID, “/path/to/XYZ” }, // This is newly added by the above
+    { App-ID, “/path/to/some/other/app/already/there” },
+    ... etc.,
+]
+```
+- The format of the config file will be CBOR (compact-binary-object-representation). The name of the local config file should be `<LOCAL-CONFIG-FILE> = launcher.config`. The config file location flowchart shall be same as that of `crust` crate's.
+
+**step 2:** User activates the app (eg., double click) from within the Launcher.
+
+**step 3:** Launcher checks the App-ID, reads the path from the `<LOCAL-CONFIG-FILE>` that it made and starts the app as an independent process. The Launcher supplies a random port on which it will listen to this app via command line options.
+
+**/path/to/app/binary --launcher "udp:<random-utf8-port>:<random-utf8-string>"**
+
+All parameters are utf8 Strings.
+
+**step 4:** Launcher will wait for a predefined time of **15** seconds for data reception on that port. If it times out it will close the socket (release its binding to it)
+
+**step 5:** App generates a random encryption asymmetric keypair - `<App-Asymm-Keys>`. Then it responds on the socket asking for Launcher to give it an `<App-Specific-Symm-Key>`, its root directory-key and SAFEDrive directory-key, which Launcher had reserved as `XYZ-Root-Dir`
+- The payload format for this request shall be CBOR encoded sturcture of the following:
+```
+struct Request {
+    launcher_string      : String, // This must be the one supplied by Launcher
+    nonce                : sodiumoxide::crypto::box_::Nonce,
+    public_encryption_key: sodiumoxide::crypto::box_::PublicKey, // from <App-Asymm-Keys>
+}
+```
+
+**step 6:** The Launcher verifies the `launcher_string` field above and generates a strong random symmetric encryption key `<App-Specific-Symm-Key>`. This is encrypted using App's `public_encrytion_key` and `nonce` above.
+
+**step 7:** Launcher gives the App what it requested.
+- The payload format for this response shall be CBOR encoded sturcture of the following,
```

Comment:
"be CBOR" -> "be a CBOR"
"sturcture" -> "structure"
"following," -> "following:"